### PR TITLE
feat: add user sign-in sign-up flows doc

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add user sign-in sign-up flows doc

### CODE REVIEW

File: docs/end-user-flows/sign-up-and-sign-in/README.mdx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+sidebar_position: 1
```
Comment (charIeszhao) [inline docs/end-user-flows/sign-up-and-sign-in/README.mdx:2]: Why is this also position 1?

File: docs/end-user-flows/sign-up-and-sign-in/README.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+---
+
+# Sign-up and sign-in
+
+Sign-in and sign-up is the core interaction process for end-users to authenticate and authorize access to client applications. As a centralized OIDC-based CIAM platform, Logto provides a universal sign-in experience for users across multiple client applications and platforms.
```
Comment (charIeszhao) [inline docs/end-user-flows/sign-up-and-sign-in/README.mdx:7]: The title says `sign-up and sign-in` but the main content starts with `sign-in and signup`. Reads a bit odd.

**Review by charIeszhao — APPROVED** [review]
Please check my comments before merging

File: docs/end-user-flows/sign-up-and-sign-in/README.mdx (hunk 2 above)
Comment (simeng-li) [reply docs/end-user-flows/sign-up-and-sign-in/README.mdx:7]: updated

File: docs/end-user-flows/sign-up-and-sign-in/README.mdx (hunk 1 above)
Comment (simeng-li) [reply docs/end-user-flows/sign-up-and-sign-in/README.mdx:2]: synced offline

**Pull request merged**